programmers/level1/키패드-누르기/곽성재.js

```diff
@@ -0,0 +1,80 @@
+function solution(numbers, hand) {
+  const answer = [];
+  let positionLeft = "*";
+  let positionRight = "#";
+  numbers.forEach((num) => {
+    switch (num) {
+      case 2:
+      case 5:
+      case 8:
+      case 0:
+        const destinationCoord = findCoord(num);
+        const leftCoord = findCoord(positionLeft);
+        const rightCoord = findCoord(positionRight);
+        const leftDist = calcDist(destinationCoord, leftCoord);
+        const rightDist = calcDist(destinationCoord, rightCoord);
+        if (leftDist > rightDist) {
+          answer.push("R");
+          positionRight = num;
+        } else if (leftDist < rightDist) {
+          answer.push("L");
+          positionLeft = num;
+        } else if (leftDist === rightDist) {
+          if (hand === "right") {
+            answer.push("R");
+            positionRight = num;
+          } else {
+            answer.push("L");
+            positionLeft = num;
+          }
+        }
+        break;
+
+      case 1:
+      case 4:
+      case 7:
+        answer.push(FIXED_KEY[num]);
+        positionLeft = num;
+        break;
+
+      case 3:
+      case 6:
+      case 9:
+        answer.push(FIXED_KEY[num]);
+        positionRight = num;
+        break;
+    }
+  });
+
+  return answer.join("");
+}
+
+// prettier-ignore
+const FIXED_KEY = { 1: "L", 4: "L", 7: "L", 3: "R", 6: "R", 9: "R" };
+const KEYPAD = [
+  [1, 2, 3],
+  [4, 5, 6],
+  [7, 8, 9],
+  ["*", 0, "#"],
+];
+
+const HAND_BOARD = {
+  right: "R",
+  left: "L",
+};
+
+const findCoord = (target) => {
+  const targetCoord = [0, 0];
+  KEYPAD.forEach((row, idx) => {
+    if (row.includes(target)) {
+      targetCoord[0] = idx;
+      return;
+    }
+  });
+  targetCoord[1] = KEYPAD[targetCoord[0]].indexOf(target);
+  return targetCoord;
+};
```

Comment:
현재 코드는 includes로 순회, indexOf로 순회라서 필요없는 순회가 있는 거 같습니다.
이중 for문으로 키패드 전체를 한 번만 순회하도록 하면 순회 횟수도 줄고 중간에 루프 탈출도 가능해서 성능상 이점이 있지 않을까 싶습니다.

```suggestion
const findCoord = (target) => {
  const KEYPAD_LENGTH = KEYPAD.length;
  const PADROW_LENGTH = KEYPAD[0].length;
  
  for (let i = 0; i < KEYPAD.length; i += 1) {
    for (let j = 0; j < KEYPAD[0].length; j += 1) {
      if (KEYPAD[i][j] === target) return [i, j];
    }
  }
};
```

Reply:
전역상수를 쓰이는 함수 내부로 넣고, for문도 이중으로 변경해서 커밋해보았습니다:))
피드백 감사합니다~!!!!!!